File: module_infrastructure/src/main/java/project/redis/screening/entity/ScreeningEntity.java
package project.redis.screening.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import project.redis.common.entity.BaseEntity;
import project.redis.movie.entity.MovieEntity;
import project.redis.theater.entity.TheaterEntity;

@Entity
@Table(name = "screening")
@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ScreeningEntity extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long screeningId;

    @Column(nullable = false)
    private LocalDateTime startedAt;

    @Column(nullable = false)
    private LocalDateTime endedAt;

    @ManyToOne
    @JoinColumn(name = "movie_id", nullable = false)  // 영화와의 관계 설정
    private MovieEntity movie;

    @ManyToOne
    @JoinColumn(name = "theater_id", nullable = false)  // 영화관과의 관계 설정
    private TheaterEntity theater;
}

File: domain/src/main/kotlin/com/hanghe/redis/screening/ScreeningEntity.kt
package com.hanghe.redis.screening

import com.hanghe.redis.BaseEntity
import com.hanghe.redis.movie.MovieEntity
import com.hanghe.redis.theater.TheaterEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalTime

@Entity(name = "screenings")
class ScreeningEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val theaterName: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id", nullable = false)
    val movie: MovieEntity,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theater_id", nullable = false)
    val theater: TheaterEntity,

    // TODO: VO 분리
    @Column(nullable = false)
    val startTime: LocalTime,

    @Column(nullable = false)
    val endTime: LocalTime,
) : BaseEntity()
